engine/crates/engine-config-builder/src/lib.rs

```diff
@@ -41,6 +41,39 @@ pub fn build_config(config: &FederatedGraphConfig, graph: FederatedGraph) -> Ver
         headers: context.headers,
         subgraph_configs,
         cache: cache_config,
+        auth: build_auth_config(config),
+    })
+}
+
+fn build_auth_config(config: &FederatedGraphConfig) -> Option<AuthConfig> {
+    config.auth.as_ref().map(|auth| {
+        let providers = auth
+            .providers
+            .iter()
+            .map(|provider| match provider {
+                AuthV2Provider::OAuth2 { name, issuer, jwt } => AuthProviderConfig::OAuth2(config::OAuth2Config {
```

Comment:
OAuth2 provider support can potentially come later, don't add this as part of this PR to keep It smaller.

Reply:
done